frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/common/GitFetchSupportImpl.java

```diff
@@ -0,0 +1,463 @@
+package com.virtuslab.gitmachete.frontend.actions.common;
+
+import static com.intellij.openapi.vcs.VcsNotifier.STANDARD_NOTIFICATION;
+import static git4idea.GitUtil.findRemoteByName;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.concurrent.CancellationException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Supplier;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import com.intellij.dvcs.MultiMessage;
+import com.intellij.dvcs.MultiRootMessage;
+import com.intellij.internal.statistic.IdeActivity;
+import com.intellij.notification.NotificationType;
+import com.intellij.openapi.progress.EmptyProgressIndicator;
+import com.intellij.openapi.progress.ProcessCanceledException;
+import com.intellij.openapi.progress.ProgressManager;
+import com.intellij.openapi.progress.util.BackgroundTaskUtil;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vcs.VcsException;
+import com.intellij.openapi.vcs.VcsNotifier;
+import com.intellij.util.concurrency.AppExecutorUtil;
+import git4idea.GitUtil;
+import git4idea.commands.Git;
+import git4idea.commands.GitAuthenticationGate;
+import git4idea.commands.GitAuthenticationListener;
+import git4idea.commands.GitImpl;
+import git4idea.commands.GitRestrictingAuthenticationGate;
+import git4idea.config.GitConfigUtil;
+import git4idea.fetch.GitFetchResult;
+import git4idea.fetch.GitFetchSupport;
+import git4idea.repo.GitRemote;
+import git4idea.repo.GitRepository;
+import io.vavr.Lazy;
+import io.vavr.Tuple;
+import io.vavr.Tuple2;
+import io.vavr.collection.HashMap;
+import io.vavr.collection.List;
+import io.vavr.collection.Map;
+import io.vavr.control.Option;
+import io.vavr.control.Try;
+import kotlin.jvm.functions.Function1;
+import lombok.AllArgsConstructor;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import com.virtuslab.gitmachete.frontend.actions.toolbar.FetchAllRemotesAction;
+import com.virtuslab.logger.IPrefixedLambdaLogger;
+import com.virtuslab.logger.PrefixedLambdaLoggerFactory;
+
+/**
+ * @deprecated This implementation is a workaround to provide features missing in pre-2020.1 versions.
```

Comment:
High-level question: what's gonna happen when we definitely want to drop support for 2019.3?

Reply:
We will be able to depend on intellij `GitFetchSupportImpl`